nodejs/eks/cluster.ts

```diff
@@ -1549,3 +1487,179 @@ export class Cluster extends pulumi.ComponentResource {
+            ...tags,
+        })),
+    }, self);
+    core.nodeGroupOptions.nodeSecurityGroup = nodeSecurityGroup;
+    core.nodeGroupOptions.clusterIngressRule = eksClusterIngressRule;
+
+    const skipDefaultNodeGroup = args.skipDefaultNodeGroup || args.fargate;
+
+    // Create the default worker node group and grant the workers access to the API server.
+    const configDeps = [core.kubeconfig];
+    let defaultNodeGroup: NodeGroupData | undefined = undefined;
+    if (!skipDefaultNodeGroup) {
+        defaultNodeGroup = createNodeGroup(name, {
+            cluster: core,
+            ...core.nodeGroupOptions,
+        }, self);
+        if (defaultNodeGroup.cfnStack) {
+            configDeps.push(defaultNodeGroup.cfnStack.id);
+        }
+    }
+
+    // Export the cluster's kubeconfig with a dependency upon the cluster's autoscaling group. This will help
+    // ensure that the cluster's consumers do not attempt to use the cluster until its workers are attached.
+    const kubeconfig = pulumi.all(configDeps).apply(([kubeconfig]) => kubeconfig);
+
+    // Export a k8s provider with the above kubeconfig. Note that we do not export the provider we created earlier
+    // in order to help ensure that worker nodes are available before the provider can be used.
+    const provider = new k8s.Provider(`${name}-provider`, {
+        kubeconfig: kubeconfig.apply(JSON.stringify),
+    }, { parent: self });
+
+    // If we need to deploy the Kubernetes dashboard, do so now.
+    if (args.deployDashboard) {
+        pulumi.log.warn("Option `deployDashboard` has been deprecated. Please consider using the Helm chart, or writing the dashboard directly in Pulumi.", core.cluster);
+        createDashboard(name, {}, self, provider);
+    }
+
+    return {
+        core,
+        clusterSecurityGroup: core.clusterSecurityGroup,
+        eksCluster: core.cluster,
+        instanceRoles: core.instanceRoles,
+        awsProvider: core.awsProvider,
+        nodeSecurityGroup,
+        eksClusterIngressRule,
+        defaultNodeGroup,
+        kubeconfig,
+        provider,
+    }
+}
+
+/**
+ * This is a variant of `Cluster` that is used for the MLC `Cluster`. We don't just use `Cluster`,
+ * because not all of its output properties are typed as `Output<T>`, which prevents it from being
+ * able to be correctly "rehydrated" from a resource reference. So we use this copy instead rather
+ * than modifying the public surface area of the existing `Cluster` class, which is still being
+ * used directly by users using the Node.js SDK. Once we move Node.js over to the generated MLC SDK,
+ * we can clean all this up. Internally, this leverages the same `createCluster` helper method that
+ * `Cluster` uses.
+ *
+ * @internal
+ */
+ export class ClusterInternal extends pulumi.ComponentResource {
+    public readonly clusterSecurityGroup!: pulumi.Output<aws.ec2.SecurityGroup>;
+    public readonly core!: pulumi.Output<pulumi.Unwrap<CoreData>>;
```

Comment:
If you look at `CoreData`, you can see its properties aren't "plain" -- they're _mostly_ all typed as `Output<T>`:

https://github.com/pulumi/pulumi-eks/blob/28745d145d53a918c681ee3ae233e67a84a287b8/nodejs/eks/cluster.ts#L128-L150

As of a couple days ago, I was using an internal copy of this interface from generating the Node.js SDK that had all plain properties.

However, I realized I could just use `pulumi.Unwrap` to get rid of all the `Output<T>`s, which means I don't have to include/maintain the internal copy of the interface.